pkg/k8s/policy.go

```diff
@@ -110,16 +107,15 @@ func ServersForResource(ctx context.Context, k8sAPI *KubernetesAPI, namespace st
 
 	results := make([]string, 0)
 
-	servers, err := k8sAPI.DynamicClient.Resource(ServerGVR).Namespace(namespace).List(ctx, metav1.ListOptions{})
+	servers, err := k8sAPI.L5dCrdClient.ServerV1beta1().Servers(namespace).List(ctx, metav1.ListOptions{})
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Failed to get serverauthorization resources: %s\n", err)
 		os.Exit(1)
 	}
 
 	for _, server := range servers.Items {
-		if sel, found, _ := unstructured.NestedMap(server.UnstructuredContent(), "spec", "podSelector"); found {
-			labelSelector := selector(sel)
-			selector, err := metav1.LabelSelectorAsSelector(&labelSelector)
+		if server.Spec.PodSelector != nil {
```

Comment:
TIOLI: could reduce nesting here by flipping the condition
```suggestion
		if server.Spec.PodSelector  == nil {
		    continue
		}
```